lib/bazel/query.go

```diff
@@ -0,0 +1,102 @@
+package bazel
+
+import (
+	"fmt"
+	"io"
+	"os/exec"
+
+	bpb "github.com/enfabrica/enkit/lib/bazel/proto"
+	"github.com/enfabrica/enkit/lib/proto/delimited"
+
+	"google.golang.org/protobuf/proto"
+)
+
+// streamedBazelCommand exec's out to bazel in the specified workspace with the
+// specified arguments. It is a variable that allows for stubbing during tests.
+//
+// In production, the implementation will return an io.Reader containing stdout,
+// an error channel that will emit any errors (if present) during execution, and
+// an error if any occur while starting the command. errChan is closed after the
+// command completes, but the caller should read all of the returned io.Reader
+// before checking the error channel.
+var streamedBazelCommand = func(cmd *exec.Cmd) (io.Reader, chan error, error) {
```

Comment:
Just adding in a comment that it would probably be best just to write into a buffer file. Would also address Carlos comments.